### ISSUE TITLE
Error: Missing target element "autocomplete.results"

### ISSUE THREAD

**Issue by AlainPilon** [afcapel/stimulus-autocomplete#57]
Hi,

using this DOM:

```
<div data-autocomplete-url-value="tags/search" data-controller="autocomplete">
    <input class="form-control mr-sm-2" data-autocomplete-target="input" placeholder="Tags" type="text">
    <input data-autocomplete-target="hidden" name="q[tags_name_in]" type="hidden">
    <ul class="list-group" data-autocomplete-target="results"></ul>
</div>
```

I am getting this error on page load: 

```
application.js:250 Error connecting controller

Error: Missing target element "autocomplete.results"
    at Controller.get (target_properties.js:15)
    at Controller.e.connect (stimulus-autocomplete.js:18)
    at Context.connect (context.js:23)
    at Module.connectContextForScope (module.js:39)
    at Router.scopeConnected (router.js:100)
    at ScopeObserver.elementMatchedValue (scope_observer.js:55)
    at ValueListObserver.tokenMatched (value_list_observer.js:54)
    at TokenListObserver.tokenMatched (token_list_observer.js:82)
    at token_list_observer.js:69
    at Array.forEach (<anonymous>)
```

Everything seems to be there and this is almost a copy&paste from the original example. Any idea what is going on?

**Comment by drnic** [comment]
I can't think of anything.

Is there any possibility you have css or js that removes that `<ul class="list-group" data-autocomplete-target="results"></ul>` element, or that it doesn't exist yet, before the controller `connect()` is run?

**Comment by AlainPilon** [comment]
Your explanation makes sense, but I can't figure out why/how the `connect` call would be run before the HTML is rendered. It is all static code. I will try on another page with a simpler setup and see how it goes. 

I will close the issue since it seems to be specific to my setup. Thx

**Comment by drnic** [comment]
I trust your problem is real; you’re getting a real error :)

If you do figure it out, please leave a note here for the next person (who
might even be you one day)

Thx
-- 
Dr Nic Williams
+61 437 276 076
twitter @drnic

**Comment by AlainPilon** [comment]
Yes the problem is real and apparently, I am not the only one having similar weird issues: 

- https://stackoverflow.com/questions/64216804/why-am-i-getting-target-element-missing-error
- https://discuss.hotwire.dev/t/missing-target-element-error/292/5

I even tried this (promising) solution:

- https://github.com/hotwired/stimulus/issues/97#issuecomment-363874711

but it did not work either. I am wondering if this is Hotwire/Turbo related.

**Comment by rafaelmadeira** [comment]
I was just having this same issue trying to implement [this auto-growing textarea](https://gist.github.com/yunusemredilber/c0598e168ef643fa8e9876b78c447b91).

It worked perfectly until I tried to submit a form with validation errors. Then the auto-growth broke and I got the `Error: Missing target element "auto-grow-textarea.input"`.

I'm using Rails 7 with Turbo and forms are submitted via AJAX so I figured the lack of actual reloading was somehow messing with the process. I'm not sure exactly what information gets persisted/reloaded during these requests, but I figured the textarea had to be part of it, so I moved the controller declaration from the parent element to the `<textarea>` element itself. It worked. In the end, it looked like `data: { controller: "auto-grow-textarea", action: ...`. 

Sharing just in case it might help others.
